extensions/notebook/resources/notebooks/JupyterBooksCreate.ipynb

```diff
@@ -129,15 +152,36 @@
                 "        if (headers):\n",
                 "            for header in headers:\n",
                 "                title = header[10:].rstrip()\n",
-                "                contentFolders.append(title.lower())\n",
-                "                filtered = list(filter(lambda x: (\"%s%s%s\" % (os.path.sep, title.lower(), os.path.sep)) in x, urls))\n",
+                "                filtered = list(filter(lambda x: (\"%s%s%s\" % (os.path.sep, title.lower(), os.path.sep)) in x.lower(), urls))\n",
                 "                index = urls.index(filtered[len(filtered)-1])\n",
+                "                markdownFilePath = getMarkdownFile(filtered[len(filtered)-1])\n",
+                "                markdownExists = path.exists(markdownFilePath)\n",
+                "                if (not markdownExists):\n",
+                "                    index = urls.index(filtered[0])\n",
+                "                folderEndIndex = urls[index].rindex(os.path.sep)\n",
+                "                caseSensitiveFolderName = urls[index][urls[index].rindex(os.path.sep, 0, folderEndIndex)+1:folderEndIndex]\n",
+                "                contentFolders.append(caseSensitiveFolderName)\n",
                 "                urlValue = urls[index][urls[index].rindex(os.path.sep)+1:].rstrip()\n",
-                "                replacedString = \"\\n- title: %s\\n  url: /%s/%s\\n  not_numbered: true\\n  expand_sections: true\\n  sections:  \" % (title, title.lower(), urlValue)\n",
+                "                replacedString = \"\\n- title: %s\\n  url: /%s/%s\\n%s  expand_sections: true\\n  sections:  \" % (title, caseSensitiveFolderName, urlValue, '  not_numbered: true\\n' if markdownExists else '')\n",
                 "                result = result.replace(header, replacedString)\n",
-                "                result = result.replace(urls[index], '')\n",
-                "                del urls[index]\n",
-                "        if (urls):\n",
+                "                if (markdownExists):\n",
+                "                    result = result.replace(urls[index], '')\n",
+                "                    del urls[index]\n",
+                "            if (urls):\n",
+                "                for url in urls:\n",
+                "                    title = url[url.rindex(os.path.sep)+1:].rstrip()\n",
```

Comment:
You're just trying to get the filename out of this right? Does path.split/path.basename not work for that? Doing manual string parsing like this isn't great...

Reply:
the prob is url is not path it's string which has data along with path. For ex: (- url: DsMain\content\Scenario1\1.MakeYourFirstCodeChange) Thus all the fuss.